tests/src/test_flow.cpp

```
@@ -0,0 +1,716 @@
+    g.layers.push_back(title);
+    g.layers.push_back(MakeMockSpec("game"));
+
+    auto flow = f.MakeFlow(std::move(g));
+    flow->Start();
+    f.log.clear();
+
+    flow->Trigger("title.play");
+    flow->Tick(0);
+
+    // The transition still completes end-to-end.
+    REQUIRE(f.log.back() == "game.In:transition_in");
```

Cppcheck (2.20.0)

[error] Out of bounds access in expression 'f.log.back()' because 'f.log' is empty.

(containerOutOfBounds)